UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs

```diff
@@ -353,24 +355,43 @@ public async Task<bool> CreateExchangeSetFile(string batchId, string correlation
             return await CreateFile(fileCreateMetaData, accessToken, correlationId);
         }
 
-        public async Task<string> CommitAndGetBatchStatus(string batchId, string correlationId, string accessToken, CustomFileInfo customFileInfo)
+        public async Task<BatchStatus> CommitAndGetBatchStatus(string batchId, string correlationId, string accessToken, CustomFileInfo customFileInfo)
         {
             BatchCommitMetaData batchCommitMetaData = new BatchCommitMetaData()
             {
                 BatchId = batchId,
                 AccessToken = accessToken,
                 FullFileName = customFileInfo.FullName
             };
-            bool isBatchCommitted = await UploadCommitBatch(batchCommitMetaData, correlationId);
-            string batchStatus = BatchStatus.CommitInProgress.ToString();
-            if (isBatchCommitted)
+            bool isUploadCommitBatchCompleted = await UploadCommitBatch(batchCommitMetaData, correlationId);
+            BatchStatus batchStatus = BatchStatus.CommitInProgress;
+            if (isUploadCommitBatchCompleted)
             {
                 var batchStatusMetaData = new BatchStatusMetaData()
                 {
                     AccessToken = accessToken,
                     BatchId = batchId
                 };
-                batchStatus = await GetBatchStatus(batchStatusMetaData, correlationId);
+                Stopwatch watch = new Stopwatch();
+                watch.Start();
+                while (batchStatus != BatchStatus.Committed && watch.Elapsed.TotalMinutes <= fileShareServiceConfig.Value.BatchCommitCutOffTimeInMinutes)
+                {                  
+                    batchStatus = await GetBatchStatus(batchStatusMetaData, correlationId);
+                    if (batchStatus == BatchStatus.Failed)
+                    {
+                        watch.Stop();
+                        logger.LogError(EventIds.BatchFailedStatus.ToEventId(), "Batch status failed for BatchId {batchId} and _X-Correlation-ID:{CorrelationId}", batchStatusMetaData.BatchId, correlationId);
+                        return batchStatus;
+                    }
+                 await Task.Delay(fileShareServiceConfig.Value.BatchCommitDelayTimeInMilliseconds);
+                }
+                if (batchStatus != BatchStatus.Committed)
+                {
+                    watch.Stop();
+                    logger.LogError(EventIds.BatchCommitTimeout.ToEventId(), "Batch Commit Status timedout with BatchStatus:{batchStatus} for BatchId:{batchId} and _X-Correlation-ID:{CorrelationId}", batchStatus, batchStatusMetaData.BatchId, correlationId);
+                    return batchStatus;
+                }
+                watch.Stop();
             }
```

Comment:
If isUploadCommitBatchCompleted is False will we return BatchStatus.CommitInProgress?

Is this correct?

Should we set BatchStatus to something more appropriate (Failed?) before returning the value?

Reply:
Resolved.